### PR TITLE
[expo-permissions] Fix permissions in the headless mode

### LINKED ISSUE #7456: Missing CAMERA_ROLL permissions when running in headless mode
### Environment

  Expo CLI 3.15.5 environment info:
    System:
      OS: Linux 4.15 Linux Mint 19.1 (Tessa)
      Shell: 5.4.2 - /usr/bin/zsh
    Binaries:
      Node: 13.9.0 - ~/.nvm/versions/node/v13.9.0/bin/node
      Yarn: 1.22.0 - ~/.nvm/versions/node/v13.9.0/bin/yarn
      npm: 6.13.7 - ~/.nvm/versions/node/v13.9.0/bin/npm
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
    npmPackages:
      react: 16.9.0 => 16.9.0 
      react-native: 0.61.5 => 0.61.5 
    npmGlobalPackages:
      expo-cli: 3.15.5

Android. Bare React Native

### Steps to Reproduce

We are trying to run some photo processing in the background when the app is closed. To do this, we are using Headless JS from React Native. This involves reading photos from the camera roll & getting AssetInfo from these.

When the app is still running and we run the BG task, everything works fine. When the app is terminated and we run the BG task, the headless task errors out.

### Expected Behavior

I expected the app to be able to load Assets in a background task. 

### Actual Behavior

The function errors out with `Error: Missing CAMERA_ROLL permissions.` 

### Reproducible Demo

https://github.com/niekcandaele/repro-expo-medialibrary-issue

You can trigger the headless task with 
`adb shell cmd jobscheduler run -f com.testappbackgroundfetch 999`

Logs when app is still running:
```
 LOG  [BackgroundFetch HeadlessTask] start:  undefined
 LOG  {"canAskAgain": true, "expires": "never", "granted": true, "status": "granted"}
 LOG  [BackgroundFetch HeadlessTask] response:  {"assets": [{Output truncated}]}
```

Logs when app is terminated:
```
 LOG  [BackgroundFetch HeadlessTask] start:  react-native-background-fetch
 LOG  {"canAskAgain": false, "expires": "never", "granted": false, "status": "denied"}
 LOG  Error getting assets from library! Error: Missing CAMERA_ROLL permissions.
```

image

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolves https://github.com/expo/expo/issues/7456.

# How

- fallback to the application context when the current activity is null.

# Test Plan

- https://github.com/niekcandaele/repro-expo-medialibrary-issue ✅
- ncl ✅

### CODE REVIEW

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedPermissionsService.kt [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
+import host.exp.exponent.di.NativeModuleDepsProvider
+import host.exp.exponent.kernel.ExperienceId
+import host.exp.exponent.kernel.services.ExpoKernelServiceRegistry
+import org.unimodules.core.ModuleRegistry
 import org.unimodules.interfaces.permissions.PermissionsResponseListener
+import javax.inject.Inject
+
+class ScopedPermissionsService(context: Context, val experienceId: ExperienceId) : PermissionsService(context) {
+
+  // This variable cannot be lateinit, cause the Location module gets permissions before this module initialized.
```
Comment (bbarthec) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedPermissionsService.kt:16]:
```suggestion
  // This variable cannot be lateinit, cause the Location module gets permissions before this module is initialized.
```

File: android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.java [hunk 2]
```diff
@@ -691,26 +690,7 @@ public void onRequestPermissionsResult(final int requestCode, final String[] per
...
-      return globalResult;
-    }
-
-    if (globalResult == PackageManager.PERMISSION_GRANTED &&
-        mExpoKernelServiceRegistry.getPermissionsKernelService().hasGrantedPermissions(permission, mExperienceId)) {
-      return PackageManager.PERMISSION_GRANTED;
-    } else {
-      return PackageManager.PERMISSION_DENIED;
-    }
+    return mExpoKernelServiceRegistry.getPermissionsKernelService().getFinalPermissions(globalResult, getPackageManager(), permission, mExperienceId);
```
Comment (bbarthec) [inline android/expoview/src/main/java/host/exp/exponent/experience/ReactNativeActivity.java:693]:
```suggestion
    return mExpoKernelServiceRegistry.getPermissionsKernelService().getPermissions(globalResult, getPackageManager(), permission, mExperienceId);
```

Why `finalPermissions`? 🤔 IS there any hidden sense I'm not getting here?

File: android/expoview/src/main/java/host/exp/exponent/kernel/services/PermissionsKernelService.java [hunk 3]
```diff
@@ -93,6 +95,32 @@ public boolean hasGrantedPermissions(String permission, ExperienceId experienceI
     return false;
   }
 
+  public int getFinalPermissions(int globalPermissionStatus, PackageManager packageManager, String permission, ExperienceId experienceId) {
```
Comment (bbarthec) [inline android/expoview/src/main/java/host/exp/exponent/kernel/services/PermissionsKernelService.java:98]: Cannot we just go with `getPermissions`? 🤔

**Review by bbarthec — COMMENTED** [review]
LGTM 😉

**Review by mczernek — APPROVED** [review]
💯

**Review by bbarthec — APPROVED** [review]

**Pull request merged**